### PR TITLE
DDA-81: Hand

### PR DESCRIPTION
**Pull request by opohjola**
Take linked diory to hand in grid lens
View diorys in hand
Link diory from hand to focus diory in grid lens
Link diory from hand to linked diory in grid lens
Link linked diory to other linked diory in grid lens

### CODE REVIEW

File: src/react/components/Draggable.js
Comment (jvalanen) [inline src/react/components/Draggable.js:15]: Did I understand right that this syntax equals to:
```
drag = useDrag({ item: {...} })[1]
```
And apparently you can do stuff like this with it:
```
const [,,,,five,,,,,ten] = [1,2,3,4,5,6,7,8,9,10]
```

These new JS syntaxes are so weird 😄

File: src/react/components/Droppable.js
Comment (jvalanen) [inline src/react/components/Droppable.js:3]: Do those indications come from `react-dnd` library? They are very good!

I was thinking that similar indications would be very nice also when using edit & delete tools (as they currently don't have any indications)... But if we want them, we have to build them by ourselves from scratch?

File: src/react/features/lenses/grid/GridLens.js
Comment (jvalanen) [inline src/react/features/lenses/grid/GridLens.js:12]: Did you already delete the useGridLens.js file? Did I remember right that you said that it's not use anywhere?

File: src/react/features/lenses/grid/GridLens.js
Comment (jvalanen) [inline src/react/features/lenses/grid/GridLens.js]: Why `focus`? Isn't that the element which is dragged from the hand and is dropped on this diory? Or at least it doesn't have anything to do with the diory in focus? 🤔 

Could it be `draggable` instead? Or `handDiory`?

File: src/react/features/lenses/grid/GridView.js
Comment (jvalanen) [inline src/react/features/lenses/grid/GridView.js:17]: Could `diorys` be renamed to `linkedDiorys`? Or `linkedDiories`? Or `linkDiorys` / `linkDiories`?

File: src/react/features/lenses/grid/GridView.js
Comment (jvalanen) [inline src/react/features/lenses/grid/GridView.js:10]: Could `dioryInFocus` be a better name for `diory`?

File: src/react/features/lenses/grid/GridView.js
Comment (jvalanen) [inline src/react/features/lenses/grid/GridView.js]: Could `droppedDioryId` be better name for `id`?

File: src/react/features/tools/hand/useHand.js
Comment (jvalanen) [inline src/react/features/tools/hand/useHand.js:11]: Could `dioryId` or `droppedDioryId` be better name for `id`? As we have also many kind of diories (diory in focus, linked diory...) and many kind of id's (or do we? link id's?)

File: src/react/features/tools/hand/HandTool.js
Comment (jvalanen) [inline src/react/features/tools/hand/HandTool.js:9]: You don't need to rename it know but I have thought about that `state.toolbarButtons` would be a better name for `state.buttons`... 🤔

File: src/react/features/lenses/grid/GridView.js
Comment (jvalanen) [inline src/react/features/lenses/grid/GridView.js:38]: This Component tree looks pretty interesting after this change:
image

File: src/react/features/tools/hand/HandTool.js
Comment (jvalanen) [inline src/react/features/tools/hand/HandTool.js:13]: Defining a new tool looks very simple!! 💪

File: src/react/features/tools/hand/handTool.feature
Comment (jvalanen) [inline src/react/features/tools/hand/handTool.feature:15]: Did you try to make a step-definition for drag? How did it go?

File: src/react/store/initialState.js
Comment (jvalanen) [inline src/react/store/initialState.js:14]: Is there something tools related in the initialState or why this was added? Just forgotten before?

**Review by jvalanen — COMMENTED** [review]
I understood surprisingly a lot of this, progress!! 😄  

Didn't dive into the React DnD details though, I will study it in more detail once I have to make first related changes to it... Seemed pretty good library though! And provides nice indications from out-of-the-box!

Now that I understood more than ever before I started to think and question the concepts and naming you have used. I made some suggestions which I think would make the code easier to read and understand. And I'll welcome you to list and decide our concepts in here: https://paper.dropbox.com/doc/Dictionary-glossary-kJ0w35JNiFlWx8skly8XH That will help us to name things and stay in-line with them.

Once you have made your changes I'd like to take a one look more. But I don't think there's need to change anything significant. But won't approve before I hear what you think about my suggestions and answer my questions :)

File: src/react/components/Draggable.js
Comment (opohjola) [reply src/react/components/Draggable.js:15]: yes

File: src/react/components/Droppable.js
Comment (opohjola) [reply src/react/components/Droppable.js:3]: isOver is a custom indicator that is defined in collect.
interface is thus: const [collect, drop] = useDrop(...

Comment (opohjola) [reply src/react/components/Droppable.js:3]: yes, maybe some <Indicator /> component could do the job

File: src/react/features/lenses/grid/GridLens.js
Comment (opohjola) [reply src/react/features/lenses/grid/GridLens.js:12]: yes, I did it earlier.

File: src/react/features/lenses/grid/GridLens.js
Comment (opohjola) [reply src/react/features/lenses/grid/GridLens.js]: GridView should be independent of hand tool, also naming, maybe we use drop for something else later. It was earlier { diory, link }, maybe that's better. And  dropped diory is not always coming from hand.

Comment (opohjola) [reply src/react/features/lenses/grid/GridLens.js]: { droppedDiory, draggedDiory }?

File: src/react/features/lenses/grid/GridView.js
Comment (opohjola) [reply src/react/features/lenses/grid/GridView.js:10]: I agree that diory and diorys is not good enough. Diory is not always in focus... e.g. when filter is ON. Also this component could / should be independent of other logic. rootDiory?

Comment (opohjola) [reply src/react/features/lenses/grid/GridView.js:10]: Other option: diorys = [rootDiory, 1linkedDiorys, 2linkedDiorys, ...]

File: src/react/features/lenses/grid/GridView.js
Comment (opohjola) [reply src/react/features/lenses/grid/GridView.js]: onDrop=({ id }) says already dropped diory id

File: src/react/features/lenses/grid/GridView.js
Comment (opohjola) [reply src/react/features/lenses/grid/GridView.js:17]: they are not always linked diorys...

File: src/react/features/lenses/grid/GridView.js
Comment (opohjola) [reply src/react/features/lenses/grid/GridView.js:38]: yes, that it is :)

File: src/react/features/tools/hand/HandTool.js
Comment (opohjola) [reply src/react/features/tools/hand/HandTool.js:9]: are they toolbar buttons? Maybe toolbar should be changed to buttonbar? :)

File: src/react/features/tools/hand/HandTool.js
Comment (opohjola) [reply src/react/features/tools/hand/HandTool.js:13]: yes, but I do not like that tool depends on lenses, it should be other way.

File: src/react/features/tools/hand/handTool.feature
Comment (opohjola) [reply src/react/features/tools/hand/handTool.feature:15]: I didn't :(

File: src/react/features/tools/hand/useHand.js
Comment (opohjola) [reply src/react/features/tools/hand/useHand.js:11]: I think diorys could have just "id"; other concepts could have specific name like lensId or filterId.
This could be improved by onDioryDrop: ({ id })

File: src/react/store/initialState.js
Comment (opohjola) [reply src/react/store/initialState.js:14]: yes, hand state is in tools

File: src/react/features/lenses/grid/GridLens.js
Comment (jvalanen) [reply src/react/features/lenses/grid/GridLens.js]: Is `link` also a diory?!

Then definitely: { droppedDiory, draggedDiory }?

File: src/react/features/lenses/grid/GridView.js
Comment (jvalanen) [reply src/react/features/lenses/grid/GridView.js:17]: What they are then?

File: src/react/features/tools/hand/HandTool.js
Comment (jvalanen) [reply src/react/features/tools/hand/HandTool.js:9]: Just suggesting what I think would be more clear to me...

File: src/react/features/lenses/grid/GridView.js
Comment (jvalanen) [reply src/react/features/lenses/grid/GridView.js:10]: Apparently I don't understand enough yet to have good suggestions... 😔

File: src/react/features/tools/hand/useHand.js
Comment (jvalanen) [reply src/react/features/tools/hand/useHand.js:11]: I don't like having generic id or data or object kind of naming. And on the other hand I don't see why explicit `dioryId` is any way worse than having just `id`.

**Review by jvalanen — APPROVED** [review]
I thought I understood something but apparently I didn't... 😔 So as we have still long road to common understanding and I got my suggestions communicated, it's best just merge this and continue conversations once I get better understanding of this all.

File: src/react/features/lenses/grid/GridView.js
Comment (opohjola) [reply src/react/features/lenses/grid/GridView.js:17]: e.g. if filter is on they are filtered diorys

**Pull request merged**
